src/Services/Settings.js

```diff
@@ -85,11 +87,45 @@ class Settings {
     // Ensure new settings variables get added by merging all the settings,
     // where the current values take precedence. 
     let currentSettings = this._getAllRaw()
+
+    // Add any non-additive settings changes here by creating a function which
+    // handles the settings change in question.
+    currentSettings = this.migrateMnemonicSettings(currentSettings);
+
     currentSettings = _.merge({}, initialSettings, currentSettings);
 
     // Apply the merged settings
     this.setAll(currentSettings);
   }
+
+  migrateMnemonicSettings(currentSettings) {
+    // If we're migrating a settings file from before we used a persistent,
+    // randomly generated mnemonic by default, randomizeMnemonic on start will
+    // be undefined.
+    if (currentSettings.server.randomizeMnemonicOnStart === undefined) {
```

Comment:
Is there something we can do here with version numbers instead to designate whether a migration is needed? I can see future migrations getting unwieldy unless they were keyed by version number. This, of course, would require adding the version number to the settings file and saving it on start.

Reply:
I think we'll need that eventually, but I figured it's best to solve the problems we have now, and leave tomorrow's problems for tomorrow.